--- src/libFLAC/bitreader.c	Tue Sep 11 05:48:56 2007
+++ src/libFLAC/bitreader.c	Tue Nov 03 19:19:37 2009
@@ -149,13 +149,23 @@
 	FLAC__CPUInfo cpu_info;
 };
 
-#ifdef _MSC_VER
+#if defined (_MSC_VER)
 /* OPT: an MSVC built-in would be better */
 static _inline FLAC__uint32 local_swap32_(FLAC__uint32 x)
 {
 	x = ((x<<8)&0xFF00FF00) | ((x>>8)&0x00FF00FF);
 	return (x>>16) | (x<<16);
 }
+
+#ifdef _WIN64
+static void local_swap32_block_(FLAC__uint32 *start, FLAC__uint32 len)
+{
+	while (len--) {
+		*start = local_swap32_(*start);
+		++start;
+	}
+}
+#else
 static void local_swap32_block_(FLAC__uint32 *start, FLAC__uint32 len)
 {
 	__asm {
@@ -173,6 +183,7 @@
 done1:
 	}
 }
+#endif
 #endif
 
 static FLaC__INLINE void crc16_update_word_(FLAC__BitReader *br, brword word)
